executor/agents/agno/mcp_manager.py

```
@@ -84,8 +79,14 @@ async def setup_mcp_tools(
                 # Connect all MCP tools in the list
                 for mcp_tool in mcp_tools_list:
                     logger.info(f"Connecting to MCP server: {mcp_tool}")
-                    await mcp_tool.connect()
-                    self.connected_tools.append(mcp_tool)
+                    try:
+                        await mcp_tool.connect()
+                        self.connected_tools.append(mcp_tool)
+                    except Exception as connect_error:
+                        logger.error(
+                            f"[MCP_CONNECT_FAIL] {type(connect_error).__name__}: {str(connect_error)}"
+                        )
```

Ruff (0.14.10)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)